### PR TITLE
Feature/pon config

### PR DESCRIPTION
**Pull request by ashwini06**
### This PR:

Creates a pon config json file for panel of normal analysis. 
With resultant config json, PON workflow for reference file generation can be executed with existing `balsamic run analysis`

Test Command: 
```
balsamic config pon --case-id securetoad10 -p  /home/proj/production/cancer/reference/target_capture_bed/production/balsamic/gmcksolid_4.1_hg19_design.bed --balsamic-cache /home/proj/long-term-stage/cancer/UMI_analysis_runs_APJ/balsamic_test_cache/ --analysis-dir /home/proj/long-term-stage/cancer/PON_analysis_runs_APJ/PON_test_datasets/ --fastq-path /home/proj/long-term-stage/cancer/PON_analysis_runs_APJ/PON_test_datasets/securetoad/fastq/ -g hg19
```

Snakemake config run screenshot:
a1

Running PON workflow : 

```
balsamic run analysis -s /home/proj/long-term-stage/cancer/PON_analysis_runs_APJ/PON_test_datasets/securetoad2/securetoad2_PON.json --account development
```

Screenshot 2021-05-03 at 07 33 15

### CODE REVIEW

**Review by hassanfa — CHANGES_REQUESTED** [review]
Some minor comments in .smk file and I think it good to go.

File: BALSAMIC/commands/config/case.py [hunk 1]
```diff
@@ -139,6 +138,7 @@ def case_config(context, case_id, umi, umi_trim_length, adapter_trim,
             "capture_kit": panel_bed,
             "chrom": get_panel_chrom(panel_bed),
         } if panel_bed else None,
+    	umiworkflow=umiworkflow if panel_bed else False
```
[the remark this answers is not in the document]
Comment (ashwini06) [reply BALSAMIC/commands/config/case.py:141]: In PON cli command, we don't have the option for umiworkflow. And in order to reuse the `AnalysisModel` in  `PonBalsamicConfigModel` I moved `umiworkflow` to BalsamicConfigModel.

[the remark this answers is not in the document]
Comment (hassanfa) [reply BALSAMIC/commands/config/case.py:141]: Makes sense. We need to refactor models as well so they can inherit properly.

File: BALSAMIC/workflows/PON.smk [hunk 2]
```diff
@@ -20,98 +19,35 @@ shell.prefix("set -eo pipefail; ")
...
-REMOVE_DUPLICATES={params.rm_dup} \
-METRICS_FILE='{output.stats}';
-samtools index {output.mrkdup};
-rm -rf {params.tmpdir};
-        """
+include: "../snakemake_rules/quality_control/fastp.rule"
+include: "../snakemake_rules/align/bwa_mem.rule"
 
+rule all:
+    input: ALL_REFS + ALL_COVS +  PON_DONE #+ CNV_DONE
```
[the remark this answers is not in the document]
Comment (ashwini06) [reply BALSAMIC/workflows/PON.smk]: Fixed!

**Review by hassanfa — APPROVED** [review]
Very nice work! good to go.

**Pull request merged**